src/main/java/org/folio/edge/sip2/MainVerticle.java

```diff
@@ -82,6 +86,28 @@ private String getSanitizedConfig() {
   public void start(Promise<Void> startFuture) {
     log.debug("Startup configuration: {}", this::getSanitizedConfig);
 
+    HttpServer httpServer = vertx.createHttpServer();
+
+    httpServer.requestHandler(request -> {
+      log.info("path : {}",request.path());
+      if (request.path().equals("/admin/health")) {
+        HttpServerResponse response = request.response();
+        response.setStatusCode(200);
+        response.putHeader("Content-Type", "text/plain");
+        response.end("OK");
+        log.info("statusCode : {}",response.getStatusCode());
+        log.info("message : {}",response.getStatusMessage());
+      }
+    });
+
+    httpServer.listen(HEALTH_CHECK_PORT, res -> {
+      if (res.succeeded()) {
+        log.info("Health endpoint is now listening!");
+      } else {
+        log.info("Failed!");
+      }
+    });
+
```

Comment:
I would have preferred this code be put in a dedicated method or a function. Also, when the health check fails should it be logged as info failed ? Can you please check the other modules how its being handled.